src/main/java/org/apache/maven/plugins/checkstyle/exec/DefaultCheckstyleExecutor.java

```diff
@@ -49,28 +52,33 @@
 import org.apache.maven.artifact.DependencyResolutionRequiredException;
 import org.apache.maven.model.Resource;
 import org.apache.maven.project.MavenProject;
-import org.codehaus.plexus.component.annotations.Component;
-import org.codehaus.plexus.component.annotations.Requirement;
 import org.codehaus.plexus.logging.AbstractLogEnabled;
 import org.codehaus.plexus.resource.ResourceManager;
 import org.codehaus.plexus.resource.loader.FileResourceCreationException;
 import org.codehaus.plexus.resource.loader.FileResourceLoader;
 import org.codehaus.plexus.resource.loader.ResourceNotFoundException;
 import org.codehaus.plexus.util.FileUtils;
+import org.eclipse.sisu.Typed;
 
 /**
  * @author Olivier Lamy
  * @since 2.5
- *
  */
-@Component(role = CheckstyleExecutor.class, hint = "default", instantiationStrategy = "per-lookup")
+@Named("default")
+@Typed(CheckstyleExecutor.class)
```

Comment:
For some reason Sisu doesn't recognize this as a possible fulfillment of CheckstyleExecutor